backend/controllers/prisonerProfile/prisonerLocationHistory.js

```diff
@@ -12,7 +12,37 @@ const {
   app: { notmEndpointUrl: dpsUrl },
 } = require('../../config')
 
-module.exports = ({ prisonApi, logError }) => async (req, res) => {
+const fetchStaffName = (context, staffId, prisonApi) => {
+  return prisonApi
+    .getStaffDetails(context, staffId)
+    .then(staff => formatName(staff.firstName, staff.lastName))
+    .catch('No staff name found')
```

Comment:
This is never used.

Reply:
This code ignores the original error and replaces it with: **No staff name found** This could be okay but looking at the code lower down you call it with an **await** This will throw an Error that is caught by the try catch block resulting in a string as the error parameter. It would also be passed into the logError function resulting more errors thrown due to that function expecting an Error object.

Reply:
I agree this looks wrong as it is now it should be `.catch(err => 'No staff name found')` and what I intend from this: is that if any errors occur during the fetching process then the staff name would be set to 'No staff name found', does that sound correct?

Reply:
That is correct. I misunderstood how it worked. However as we've discussed it's probably best to just let the error bubble up so that it gets logged correctly.